lib/spack/llnl/util/filesystem.py

```diff
@@ -211,7 +212,8 @@ def copy_mode(src, dest):
         dest_mode |= stat.S_IXGRP
     if src_mode & stat.S_IXOTH:
         dest_mode |= stat.S_IXOTH
-    os.chmod(dest, dest_mode)
+    if not os.path.islink(dest):
```

Comment:
Could you return at the beginning of the function if this is a link? It demonstrates that none of the above logic is necessary when `dest` is a symlink.